org.eclipse.scout.rt.mail/src/main/java/org/eclipse/scout/rt/mail/smtp/SmtpConnectionPool.java

```diff
@@ -320,17 +351,22 @@ protected void releaseConnection(SmtpConnectionPoolEntry poolEntry) {
    *         provided entry.
    */
   protected SmtpConnectionPoolEntry exchangeConnection(SmtpConnectionPoolEntry oldEntry) throws MessagingException {
+    SmtpServerConfig smtpServerConfig = oldEntry.getSmtpServerConfig();
+
     synchronized (m_poolLock) {
+      checkAndThrowIfDestroyed();
 
       m_leasedEntries.remove(oldEntry);
       safeCloseTransport(oldEntry);
 
-      SmtpConnectionPoolEntry newEntry = createNewConnection(oldEntry.getSmtpServerConfig());
-
-      m_idleEntries.remove(newEntry);
-      m_leasedEntries.add(newEntry);
+      m_currentlyConnectingCounter.compute(smtpServerConfig, (k, v) -> v == null ? 1 : (v + 1)); // increment
+    }
 
-      return newEntry;
+    try {
+      return createNewConnectionWithoutPoolLockAndLease(smtpServerConfig);
+    }
+    finally {
+      m_currentlyConnectingCounter.compute(smtpServerConfig, (k, v) -> v == 1 ? null : (v - 1)); // decrement (remove if 0)
```

Comment:
not sure if we should put the increment/decrements in own methods?

Reply:
Agree, thought about it also - changed in 231195a99e67f0e19d31639e53a5fd6b7b791174